base/sparse/linalg.jl

```diff
@@ -805,17 +803,67 @@ function kron(x::SparseVector{Tv,Ti},y::SparseVector{Tv,Ti}) where {Tv,Ti}
     return SparseVector(x.n*y.n,nzind,nzval)
 end
 
-function kron(x::SparseVector{Tv1,Ti1}, y::SparseVector{Tv2,Ti2}) where {Tv1,Ti1,Tv2,Ti2}
-    Tv_res = promote_type(Tv1, Tv2)
-    Ti_res = promote_type(Ti1, Ti2)
-    x2 = convert(SparseVector{Tv_res,Ti_res}, x)
-    y2 = convert(SparseVector{Tv_res,Ti_res}, y)
-    return kron(x2,y2)
-end
-
 kron(x::SparseVector{Tv,Ti}, y::AbstractVector) where {Tv,Ti} = kron(x, sparse(y))
 kron(x::AbstractVector, y::SparseVector{Tv,Ti}) where {Tv,Ti} = kron(sparse(x), y)
 
+# kron - sparse outer product case
+
+# avoid adding new methods to standard nonzeros/nonzeroinds
+_outer_nonzeros(v::SparseVectorUnion) = nonzeros(v)
+_outer_nonzeroinds(v::SparseVectorUnion) = nonzeroinds(v)
+_outer_nonzeros(v::DenseVector) = v
+_outer_nonzeroinds(v::DenseVector) = indices(v, 1)
+_outer_nonzeros(v::ConjVector) = conj(nonzeros(parent(v)))
+_outer_nonzeroinds(v::ConjVector) = nonzeroinds(parent(v))
+
+function _sparse_outer(A, B)
```

Comment:
Do you think it would make sense to do this as part of a sparse `lowrankupdate!` function?

Reply:
Sure, I can look into it. I'll take a stab at rearranging/renaming the current code to something like ```lowrankupdate(::SparseVectorUnion, ::SparseVectorUnion)``` (I'm selfish and really want the base case available to me :-P) and can work on figuring out ```lowrankupdate!```.

Reply:
I don't think it is necessary to rename the function if you don't rewrite the implementation to update an existing matrix. Just thought it would be nice to make it fit into the existing methods.

Reply:
If there's a way to do `lowrankupdate!(A, u, v)` more efficiently than `A + kron(u, v')`, I'd love to take advantage of it. For the sake of my own research progress, just getting `kron(u, v')` efficient is sufficiently low hanging fruit to help me immensly, but I won't reject even further improvements.